x-pack/legacy/plugins/siem/public/components/generic_downloader/index.tsx

```diff
@@ -28,26 +38,35 @@ export interface RuleDownloaderProps {
  * @param payload Rule[]
  *
  */
-export const RuleDownloaderComponent = ({
+
+export const GenericDownloaderComponent = ({
+  exportSelectedData,
   filename,
-  ruleIds,
-  onExportComplete,
-}: RuleDownloaderProps) => {
+  ids,
+  onExportSuccess,
+  onExportFailure,
+}: GenericDownloaderProps) => {
   const anchorRef = useRef<HTMLAnchorElement>(null);
   const [, dispatchToaster] = useStateToaster();
 
   useEffect(() => {
     let isSubscribed = true;
     const abortCtrl = new AbortController();
 
-    async function exportData() {
-      if (anchorRef && anchorRef.current && ruleIds != null && ruleIds.length > 0) {
+    const exportData = async () => {
+      if (
+        anchorRef &&
+        anchorRef.current &&
+        ((ids != null && ids.length > 0) || (ids != null && ids.length > 0))
+      ) {
+        let exportResponse;
         try {
-          const exportResponse = await exportRules({
-            ruleIds,
-            signal: abortCtrl.signal,
-          });
-
+          if (!isNil(exportSelectedData)) {
```

Comment:
why do we need the condition here? it seems like this function should always be there.